tt_metal/common/core_descriptor.hpp

```diff
@@ -106,37 +106,37 @@ inline CoreCoord get_compute_grid_size(chip_id_t device_id, const uint8_t num_hw
 
 inline const std::vector<CoreCoord> &get_logical_compute_cores(chip_id_t device_id, const uint8_t num_hw_cqs) {
     const core_descriptor_t &core_desc = get_core_descriptor_config(device_id, num_hw_cqs);
-    static std::unordered_map<chip_id_t, std::vector<CoreCoord>> logical_compute_cores_by_device;
-    if (logical_compute_cores_by_device.count(device_id)) {
-        return logical_compute_cores_by_device.at(device_id);
+    static std::unordered_map<chip_id_t, std::unordered_map<uint8_t, std::vector<CoreCoord>>> logical_compute_cores_by_device;
+    if (logical_compute_cores_by_device[device_id].count(num_hw_cqs)) {
```

Comment:
Can you use `at` for maps?

Reply:
I chose to use `[]` to default construct the device map if it didn't exist. Otherwise it would look something like `logical_compute_cores_by_device.count(device_id) && logical_compute_cores_by_device.at(device_id).count(num_hw_cqs)`.

Seemed cleaner to just have it default construct.